posthog/api/exports.py

```diff
@@ -80,6 +85,32 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> ExportedAss
 
         instance.refresh_from_db()
 
+        insight_id = instance.insight_id
+        dashboard_id = instance.dashboard_id
+        if insight_id and not dashboard_id:  # we don't log dashboard activity ¯\_(ツ)_/¯
+            try:
+                insight: Insight = Insight.objects.select_related("team__organization").get(id=insight_id)
+                log_activity(
+                    organization_id=insight.team.organization.id,
+                    team_id=self.context["team_id"],
+                    user=request.user,
+                    item_id=insight_id,  # Type: ignore
+                    scope="Insight",
+                    activity="exported",
+                    detail=Detail(
+                        name=insight.name if insight.name else insight.derived_name,
```

Comment:
Not a blocker but I see this used so much I wonder if it makes sense as a model property...

Reply:
To be clear I was referring to the `insight.name || insight.derived_name` bit of code that is everywhere